File: frontend/src/types/market.ts
export interface MarketPublic {
  id: string;
  name: string;
  description?: string;
  contact_email: string;
  contact_phone?: string;
  address?: string;
  created_at: string;
  updated_at: string;
}

export interface MarketCreate {
  name: string;
  description?: string;
  contact_email: string;
  contact_phone?: string;
  address?: string;
}

export interface MarketUpdate {
  name?: string;
  description?: string;
  contact_phone?: string;
  address?: string;
} 

File: backend/app/models/market.py
from pydantic import BaseModel, Field, EmailStr
from typing import Optional
from datetime import datetime


class MarketBase(BaseModel):
    name: str = Field(..., max_length=200)
    description: Optional[str] = None
    contact_email: EmailStr
    contact_phone: Optional[str] = None
    address: Optional[str] = None


class MarketCreate(MarketBase):
    owner_id: str


class MarketInDB(MarketBase):
    id: str
    owner_id: str
    created_at: datetime
    updated_at: datetime


class MarketPublic(MarketBase):
    id: str
    created_at: datetime
    updated_at: datetime


class MarketUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    contact_phone: Optional[str] = None
    address: Optional[str] = None 